kernel/src/syscall/prlimit64.rs

```diff
@@ -6,6 +6,27 @@ use crate::{
     process::{Pid, ResourceType},
 };
 
+pub fn sys_getrlimit(resource: u32, rlim_addr: Vaddr, ctx: &Context) -> Result<SyscallReturn> {
+    let resource = ResourceType::try_from(resource)?;
+    debug!("resource = {:?}, rlim_addr = 0x{:x}", resource, rlim_addr);
+    let resource_limits = ctx.process.resource_limits().lock();
+    let rlimit = resource_limits.get_rlimit(resource);
+    ctx.get_user_space().write_val(rlim_addr, rlimit)?;
+    Ok(SyscallReturn::Return(0))
+}
+
+pub fn sys_setrlimit(resource: u32, new_rlim_addr: Vaddr, ctx: &Context) -> Result<SyscallReturn> {
+    let resource = ResourceType::try_from(resource)?;
+    debug!(
+        "resource = {:?}, new_rlim_addr = 0x{:x}",
+        resource, new_rlim_addr
+    );
+    let mut resource_limits = ctx.process.resource_limits().lock();
+    let new_rlimit = ctx.get_user_space().read_val(new_rlim_addr)?;
```

Comment:
It's better to replace the order of line 24 and line 25. If `read_val` fails, then there's no need to acquire lock.

Reply:
Good idea!